src/Stratis.Bitcoin/BlockPulling2/BlockPuller.cs

```diff
@@ -405,25 +408,23 @@ public void RequestBlocksDownload(List<ChainedHeader> headers)
                 {
                     emptySlots = this.maxBlocksBeingDownloaded - this.assignedDownloadsByHash.Count;
                 }
-
-                this.logger.LogTrace("There are {0} empty slots.", emptySlots);
```

Comment:
Why do we remove logs? Is it because we are inside of lock? If so and there is no other reason beyond that, please revert change

Reply:
Fixed it in a way that should be good enough for you and Dan. Logging will happen only in case queue is not processed because threshold was not met. 
If it was met we will log `emptySlots` in the entry log for ProcessQueueLocked anyway so all the important data will always be logged.